src/ext/NetFx/ca/netfxca.cpp

```diff
@@ -879,3 +907,132 @@ extern "C" UINT __stdcall ExecNetFx(
     return WcaFinalize(er);
 }
 
+/******************************************************************
+ DotNetCompatibilityCheck - entry point for NetFx Custom Action
+
+*******************************************************************/
+extern "C" UINT __stdcall DotNetCompatibilityCheck(
+    __in MSIHANDLE hInstall
+    )
+{
+//    AssertSz(FALSE, "debug DotNetCompatibilityCheck");
+
+    HRESULT hr = S_OK;
+    UINT er = ERROR_SUCCESS;
+
+    DWORD dwTempPathLength = 0;
+    LPWSTR pwzTempPath = NULL;
+    PMSIHANDLE hView = NULL;
+    PMSIHANDLE hRec = NULL;
+    LPWSTR pwzPlatform = NULL;
+    LPWSTR pwzNetCoreCheckBinaryId = NULL;
+    LPWSTR pwzNetCoreCheckFilePath = NULL;
+    LPWSTR pwzRuntimeType = NULL;
+    LPWSTR pwzVersion = NULL;
+    LPWSTR pwzRollForward = NULL;
+    LPWSTR pwzVariable = NULL;
+    LPWSTR pwzCommandLine = NULL;
+
+    // Initialize
+    hr = WcaInitialize(hInstall, "DotNetCompatibilityCheck");
+    ExitOnFailure(hr, "failed to initialize");
+
+    // Get temp directory path
+    dwTempPathLength = GetTempPathW(dwTempPathLength, NULL);
+    if (0 == dwTempPathLength)
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        ExitOnFailure(hr, "failed to get temp path length");
+    }
+
+    hr = StrAlloc(&pwzTempPath, ++dwTempPathLength);
+    ExitOnFailure(hr, "failed to allocate temp path variable");
+
+    dwTempPathLength = GetTempPathW(dwTempPathLength, pwzTempPath);
+    if (0 == dwTempPathLength)
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        ExitOnFailure(hr, "failed to get temp path value");
+    }
+
+    // Open view on .NET compatibility check table
+    hr = WcaOpenExecuteView(vscDotNetCompatibilityCheckQuery, &hView);
+    ExitOnFailure(hr, "failed to open view on Wix4NetFxDotNetCompatibilityCheck table");
+
+    // Go through all records and run NetCorCheck.exe for each
+    while (S_OK == (hr = WcaFetchRecord(hView, &hRec)))
+    {
+        HANDLE hProcess = INVALID_HANDLE_VALUE;
+        DWORD dwExitCode = 0;
+
+        // Extract NetCoreCheck.exe for platform to temp directory
+        hr = WcaGetRecordString(hRec, platform, &pwzPlatform);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.Platform");
+
+        hr = GetNetCoreCheckBinaryId(pwzPlatform, &pwzNetCoreCheckBinaryId);
+        ExitOnFailure(hr, "failed to get NetCoreCheck binary id for platform %ls", pwzPlatform);
+
+        hr = StrAllocFormatted(&pwzNetCoreCheckFilePath, L"%ls%ls", pwzTempPath, L"NetCoreCheck.exe");
```

Comment:
This could be precalculated outside the loop with this implementation.

I don't know if it's worth it, but there's a potential optimization here to extract each architecture to a different path so that the same file doesn't have to be extracted more than once. Which would mean this couldn't be precalculated anymore.

Reply:
I 💯 agree, but since the most likely case is that this table has one row it isn't likely to be noticeable. This will be fine.